app/scripts/lib/rpc-method-middleware/createMethodMiddleware.js

```diff
@@ -33,14 +34,23 @@ const expectedHookNames = Array.from(
 export function createMethodMiddleware(hooks) {
   // Fail immediately if we forgot to provide any expected hooks.
   const missingHookNames = expectedHookNames.filter(
-    (hookName) => !Object.hasOwnProperty.call(hooks, hookName),
+    (hookName) => !hasProperty(hooks, hookName),
   );
   if (missingHookNames.length > 0) {
     throw new Error(
       `Missing expected hooks:\n\n${missingHookNames.join('\n')}\n`,
     );
   }
 
+  const extraneousHookNames = Object.keys(hooks).filter(
```

Comment:
Would this be more appropriate?
```suggestion
  const extraneousHookNames = Object.getOwnPropertyNames(hooks).filter(
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyNames

Reply:
I don't think we have reason to expect keys on the objects in question to be non-enumerable, but sure, why not! Done.